zephyr/drivers/can/can_sccan.c

```diff
@@ -342,6 +530,10 @@ static int sc_can_init(const struct device *dev)
 	uint32_t v;
 
 	k_mutex_init(&data->inst_mutex);
+	k_sem_init(&data->tx_sem, SCCAN_TX_FIFO_DEPTH, SCCAN_TX_FIFO_DEPTH);
+
+	data->tx_head = 0;
+	data->tx_tail = 0;
```

Comment:
Why do you initialize them?  If you do, why don't you initialize other members? Especially the ones you check against `NULL`.

Reply:
I will initialize with other members in SCCAN_INIT.